File: src/main/java/net/flyingfishflash/loremlist/domain/lrmitem/LrmItemRepository.java
package net.flyingfishflash.loremlist.domain.lrmitem;

import java.util.List;
import java.util.Set;
import java.util.UUID;

public interface LrmItemRepository {
  long countByOwner(String owner);

  int delete();

  int deleteById(Set<UUID> ids);

  int deleteByOwnerAndId(UUID id, String owner);

  List<LrmItem> findByOwner(String owner);

  LrmItem findByOwnerAndIdOrNull(UUID id, String owner);

  List<LrmItem> findByOwnerAndHavingNoListAssociations(String owner);

  List<LrmItem> findByOwnerAndHavingNoListAssociations(String owner, UUID listId);

  List<UUID> findIdsByOwnerAndIds(List<UUID> itemIdCollection, String owner);

  Set<UUID> notFoundByOwnerAndId(List<UUID> itemIdCollection, String owner);

  UUID insert(LrmItem lrmItem);

  int update(LrmItem lrmItem);

  int updateName(LrmItem lrmItem);

  int updateDescription(LrmItem lrmItem);
}

File: src/main/kotlin/net/flyingfishflash/loremlist/domain/lrmitem/LrmItemRepository.kt
package net.flyingfishflash.loremlist.domain.lrmitem

import java.util.UUID

interface LrmItemRepository {
  fun countByOwner(owner: String): Long
  fun delete(): Int
  fun deleteById(ids: Set<UUID>): Int
  fun deleteByOwnerAndId(id: UUID, owner: String): Int
  fun findByOwner(owner: String): List<LrmItem>
  fun findByOwnerAndIdOrNull(id: UUID, owner: String): LrmItem?
  fun findByOwnerAndHavingNoListAssociations(owner: String): List<LrmItem>
  fun findIdsByOwnerAndIds(itemIdCollection: List<UUID>, owner: String): List<UUID>
  fun notFoundByOwnerAndId(itemIdCollection: List<UUID>, owner: String): Set<UUID>
  fun insert(lrmItem: LrmItem): UUID
  fun update(lrmItem: LrmItem): Int
  fun updateName(lrmItem: LrmItem): Int
  fun updateDescription(lrmItem: LrmItem): Int
}
